docs/Lineage-API.md

```diff
@@ -0,0 +1,150 @@
+```
+
+### Creating a Lineage Record
+
+Lineage can be created by calling
+`TachyonLineage#createLineage(List<TachyonURI>, List<TachyonURI>, Job)`. A lineage record takes (1)
+a list of URIs of the input files, (2) a list of URIs of the output files, and (3) a *job*. A job
+is description of a program that can be run by Tachyon to recompute the output files given the input 
+files. *Note: In the current alpha version, only a built-in `CommandLineJob` is supported, which 
+simply takes a command String that can be run in a terminal. The user needs to provide the necessary 
+configurations and execution environments to ensure the command can be executed both at the client 
+and at Tachyon master (during recomputation).*
+
+For example,
+
+```java
+TachyonLineage tl = TachyonLineage.get();
+// input file paths
+TachyonURI input1 = new TachyonURI("/inputFile1");
+TachyonURI input2 = new TachyonURI("/inputFile2");
+List<TachyonURI> inputFiles = Lists.newArrayList(input1, input2);
+// output file paths
+TachyonURI output = new TachyonURI("/outputFile");
+List<TachyonURI> outputFiles = Lists.newArrayList(output);
+// command-line job
+JobConf conf = new JobConf("/tmp/recompute.log");
+CommandLineJob job = new CommandLineJob("my-spark-job.sh", conf);
+long lineageId = tl.createLineage(inputFiles, outputFiles, job);
+```
+
+The `createLineage` function returns the id of the newly created lineage record. Before creating a
+lineage record, make sure that all the input files are either persisted, or specified as an output 
+file in another lineage record.
+
+### Specifying Operation Options
+
+For all `TachyonLineage` operations, an additional `options` field may be specified, which allows
+users to specify non-default settings for the operation.
+
+### Deleting a Lineage
+
+Lineage records can be deleted by calling `TachyonLineage#deleteLineage`. The deletion function
+takes the lineage id.
+
+```java
+TachyonLineage tl = TachyonLineage.get();
+...
+long lineageId = tl.createLineage(inputFiles, outputFiles, job);
+...
+tl.deleteLineage(lineageId);
+```
+
+By default, the lineage record to delete cannot have output files depended on by other lineages.
+Optionally, all the downstream lineages can be deleted altogether by setting the cascade delete
+flag. For example:
+
+```java
+TachyonLineage tl = TachyonLineage.get();
+DeleteLineageOptions options =
+    new DeleteLineageOptions.Builder(new TachyonConf()).setCascade(true).build();
+tl.deleteLineage(1, options);
+```
+
+# Configuration Parameters For Lineage
+
+These are the configuration parameters related to Tachyon's lineage feature.
+
+<table class="table-striped">
+<tr><th>Parameter</th><th>Default Value</th><th>Description</th></tr>
+<tr>
+  <td>tachyon.master.lineage.checkpoint.interval.ms</td>
+  <td>600000</td>
+  <td>
+  The interval (in milliseconds) between Tachyon's checkpoint scheduling.
+  </td>
+</tr>
+<tr>
+  <td>tachyon.master.lineage.checkpoint.strategy.class</td>
```

Comment:
these long property names and values make the resulting table look awkward ... not sure what to do about this